Structure_Engine/Query/ReinforcementLayout.cs

```diff
@@ -401,20 +401,20 @@ private static void ExtractInnerAndOuterEdges(ConcreteSection section, out List<
 
         private static List<LongitudinalReinforcement> LongitudinalReinforcement(this ConcreteSection section)
         {
-            if (section == null || section.Reinforcement == null || section.Reinforcement.Count == 0)
+            if (section == null || section.RebarIntent == null || section.RebarIntent.BarReinforcement.Count == 0)
                 return new List<LongitudinalReinforcement>();
 
-            return section.Reinforcement.Where(x => x is LongitudinalReinforcement).Cast<LongitudinalReinforcement>().ToList();
+            return section.RebarIntent.BarReinforcement.Where(x => x is LongitudinalReinforcement).Cast<LongitudinalReinforcement>().ToList();
         }
 
         /***************************************************/
 
         private static List<TransverseReinforcement> TransverseReinforcement(this ConcreteSection section)
         {
-            if (section == null || section.Reinforcement == null || section.Reinforcement.Count == 0)
+            if (section == null || section.RebarIntent == null || section.RebarIntent.BarReinforcement.Count == 0)
```

Comment:
Check the inner list for null as well `section.RebarIntent.BarReinforcement == null` before checking the count